### PR TITLE
[MO] - [system test] - ST for support SASL-scramsha in ConnectS2I

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change

- Enhancement / new feature

### Description

This PR adds
 - ST for support SASL with mechanism scram-sha using TLS encryption

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 1]
```diff
@@ -19,42 +19,114 @@
...
-        StUtils.waitTillDepConfigHasRolled(CONNECT_DEPLOYMENT_NAME, 1, connectSnapshot);
+        StUtils.waitTillDepConfigHasRolled(CLUSTER_NAME + "-connect", 1, connectSnapshot);
         String connectS2IPodName = kubeClient().listPods("type", "kafka-connect-s2i").get(0).getMetadata().getName();
         String plugins = cmdKubeClient().execInPod(connectS2IPodName, "curl", "-X", "GET", "http://localhost:8083/connector-plugins").out();
 
         assertThat(plugins, containsString("io.debezium.connector.mongodb.MongoDbConnector"));
     }
 
+    @Test
+    @Tag(NODEPORT_SUPPORTED)
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Should this be `@OpenShiftOnly`?

Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Correct, fixed.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 2]
```diff
@@ -14,47 +14,119 @@
...
-    void testDeployS2IWithMongoDBPlugin() throws IOException {
+    void testDeployS2IWithMongoDBPlugin() {
         testClassResources().kafkaConnectS2I(CONNECT_CLUSTER_NAME, 1, CLUSTER_NAME)
             .editMetadata()
                 .addToLabels("type", "kafka-connect-s2i")
             .endMetadata()
             .done();
 
-        Map<String, String> connectSnapshot = StUtils.depConfigSnapshot(CONNECT_DEPLOYMENT_NAME);
+        Map<String, String> connectSnapshot = StUtils.depConfigSnapshot(CLUSTER_NAME + "-connect");
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]:
```suggestion
        Map<String, String> connectSnapshot = KafkaConnectS2IResources.deploymentName(CLUSTER_NAME);
```

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 3]
```diff
@@ -14,47 +14,119 @@
...
 
         File dir = StUtils.downloadAndUnzip("https://repo1.maven.org/maven2/io/debezium/debezium-connector-mongodb/0.7.5/debezium-connector-mongodb-0.7.5-plugin.zip");
 
         // Start a new image build using the plugins directory
-        cmdKubeClient().exec("oc", "start-build", CONNECT_DEPLOYMENT_NAME, "--from-dir", dir.getAbsolutePath(), "-n", NAMESPACE);
+        cmdKubeClient().exec("oc", "start-build", CLUSTER_NAME + "-connect", "--from-dir", dir.getAbsolutePath(), "-n", NAMESPACE);
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Again , use `KafkaConnectS2IResources`

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 4]
```diff
@@ -14,47 +14,119 @@
...
         // Wait for rolling update connect pods
-        StUtils.waitTillDepConfigHasRolled(CONNECT_DEPLOYMENT_NAME, 1, connectSnapshot);
+        StUtils.waitTillDepConfigHasRolled(CLUSTER_NAME + "-connect", 1, connectSnapshot);
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: `KafkaConnectS2IResources`

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 5]
```diff
@@ -14,47 +14,119 @@
...
+        testMethodResources().kafkaConnectS2I(CONNECT_CLUSTER_NAME, 1, CLUSTER_NAME)
+                .editMetadata()
+                    .addToLabels("type", "kafka-connect-s2i")
+                .endMetadata()
+                .editSpec()
+                    .addToConfig("key.converter.schemas.enable", false)
+                    .addToConfig("value.converter.schemas.enable", false)
+                    .withNewTls()
+                        .addNewTrustedCertificate()
+                            .withSecretName(CLUSTER_NAME + "-cluster-ca-cert")
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: There's a method on `KafkaResources` encapsulating this naming.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 6]
```diff
@@ -14,47 +14,119 @@
...
+                            .withCertificate("ca.crt")
+                        .endTrustedCertificate()
+                    .endTls()
+                    .withBootstrapServers(CLUSTER_NAME + "-kafka-bootstrap:9093")
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Again `KafkaResources` should have a method for this.

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectS2IST

**Comment by Seequick1** [comment]
I tried on my `local machine` 4-times and it's passed `4/4`. Also i tried it on remote machine and it was a flaky `2/4`.

**Comment by Frawless** [comment]
issues with teardown between tests will be addressed in https://github.com/strimzi/strimzi-kafka-operator/pull/2026

**Pull request merged**